extensions/rich_text_components/Video/directives/oppia-noninteractive-video.component.ts

```diff
@@ -66,7 +66,15 @@ export class NoninteractiveVideo implements OnInit {
     private htmlEscaperService: HtmlEscaperService
   ) {}
 
-  ngOnInit(): void {
+  private _updateViewOnNewVideo(): void {
+    if (
+      !this.autoplayWithValue ||
+      !this.endWithValue ||
+      !this.startWithValue ||
+      !this.videoIdWithValue
+    ) {
```

Comment:
Why do we have this if here and also the opposite if in `ngOnChanges`. Ditto in other places.

Reply:
We don't have the literal opposite in ngOnChanges even though it might look opposite. When ckEditor first creates these components, everything is undefined. All of these values are not defined and this leads to errors in the console. This previously didn't happen because we would compile templates on the fly using $compile. $compile is something that the migration team is trying to get rid off. Angular compiler is really heavy. What I have done instead is make this into a web component. (So basically, you can create this component using document.createElement. So ckeditor adds this before I can set attributes. This is we are doing this complicated thing. I will create a README at the root of the rte folder explaining this process. I am opting for a README instead of a comment because the comment will have to repeat in all places.

Coming back to condition, angular provides us with this lifecycle hook called ngOnChanges. This hook runs whenever any `@Input` decorated class property changes. We get an SimpleChanges object. https://angular.io/api/core/SimpleChanges

`changes.propname` is defined when the property changes. Which is why I checking if the following `n` properties I care about have changed.

Reply:
Okay, makes sense. README sounds nice, although should we maybe just make a small comment in all the relevant places referencing the README?

Reply:
For now I have added a comment in each file. I didn't know if every dev would go through the readme or not. Example comment: https://github.com/oppia/oppia/pull/12881/files#diff-93458cd0b245e980dbcf3a74e5b588bf45bf86e48fa26d556e85cec868c98988R21-R33. I have done that for each file. PTAL!